src/posix/platform/infra_if.cpp

```diff
@@ -91,13 +91,15 @@ bool otPlatInfraIfHasAddress(uint32_t aInfraIfIndex, const otIp6Address *aAddres
     return ret;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
 otError otPlatInfraIfSendIcmp6Nd(uint32_t            aInfraIfIndex,
                                  const otIp6Address *aDestAddress,
                                  const uint8_t      *aBuffer,
                                  uint16_t            aBufferLength)
 {
     return ot::Posix::InfraNetif::Get().SendIcmp6Nd(aInfraIfIndex, *aDestAddress, aBuffer, aBufferLength);
 }
+#endif // OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
```

Comment:
We can consider removing `// OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` when the number of lines from related `#if` is less than 20 (we have this suggestion in our style guide which can help declutter the code).